odl/discr/discr_utils.py

```diff
@@ -565,7 +565,11 @@ def __call__(self, x, out=None):
             Interpolated values. If ``out`` was given, the returned
             object is a reference to it.
         """
-        x = np.asarray(x)
+        if isinstance(x, tuple):
```

Comment:
Perhaps worth replacing by: if self.input_type == 'array':
as in line 576

Reply:
sorry, here it should be  'meshgrid'

Reply:
Honestly I can't see a real reason why there is a `self.input_type` at all and this isn't just always inferred from the input type to `__call__` instead.
On the other hand, having tuples signify meshgrid-input is definitely very obscure and prone to confusing errors, so you're right that this is not a good way to do it. Will change as suggested, and add some assertions instead.